### PR TITLE
Fix sprite rounding issues

### PR DESCRIPTION
**Pull request by metalgearsloth**
Should fix all of the issues at exactly 45 degrees

### CODE REVIEW

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 1]
```diff
@@ -1321,7 +1328,7 @@ private void RenderLayer(DrawingHandleWorld drawingHandle, Layer layer, Angle ey
         public static Angle CalcRectWorldAngle(Angle worldAngle, int numDirections)
         {
             var theta = worldAngle.Theta;
-            var segSize = (MathF.PI * 2) / (numDirections * 2);
+            var segSize = (Math.PI * 2) / (numDirections * 2);
```
Comment (metalgearsloth) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs:1330]: This is what fixes walls

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 2]
```diff
@@ -1270,6 +1270,12 @@ private void CalcModelMatrix(int numDirs, Angle eyeRotation, Angle worldRotation
 
         private void RenderInternal(DrawingHandleWorld drawingHandle, Angle eyeRotation, Angle worldRotation, Vector2 worldPosition, Direction? overrideDirection)
         {
+            // Reduce the angles to fix math shenanigans
+            worldRotation = worldRotation.Reduced();
+
+            if (worldRotation.Theta < 0)
+                worldRotation = new Angle(worldRotation.Theta + Math.Tau);
```
Comment (metalgearsloth) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs:1277]: This fixes stuff like lights

File: Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs [hunk 3]
```diff
@@ -1291,7 +1297,8 @@ private void RenderInternal(DrawingHandleWorld drawingHandle, Angle eyeRotation,
 
         private void RenderLayer(DrawingHandleWorld drawingHandle, Layer layer, Angle eyeRotation, Angle worldRotation, Direction? overrideDirection)
         {
-            var texture = GetRenderTexture(layer, worldRotation + eyeRotation, overrideDirection);
+            var rotation = worldRotation + eyeRotation;
+            var texture = GetRenderTexture(layer, rotation, overrideDirection);
```
Comment (PaulRitter) [inline Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs]: wait why did you just pull the statement out of the function call and nothing else?

**Review by PaulRitter — APPROVED** [review]
lgtm but pls ex:b:lain

**Pull request merged**